### PR TITLE
Fix radioacesstechnology api deprecated for ios 12.0 or newer

### PR DESCRIPTION
**Pull request by doudounan**
Fix radioacesstechnology api deprecated for ios 12.0 or newer

### CODE REVIEW

**Comment by paulb777** [comment]
We don't yet have automated tests for Catalyst, so please test Catalyst manually.

File: GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m [hunk 1]
```diff
@@ -71,7 +71,16 @@ GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() {
     };
     networkInfo = [[CTTelephonyNetworkInfo alloc] init];
   });
-  NSString *networkCurrentRadioAccessTechnology = networkInfo.currentRadioAccessTechnology;
+  NSString *networkCurrentRadioAccessTechnology;
+  if (@available(iOS 12, *)) {
+    NSDictionary<NSString *, NSString *> *networkCurrentRadioAccessTechnologyDict =
+        networkInfo.serviceCurrentRadioAccessTechnology;
+    if (networkCurrentRadioAccessTechnologyDict) {
+      networkCurrentRadioAccessTechnology = networkCurrentRadioAccessTechnologyDict.allValues[0];
```
Comment (morganchen12) [inline GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m]: Based on the [documentation for this method](https://developer.apple.com/documentation/coretelephony/cttelephonynetworkinfo/3024510-servicecurrentradioaccesstechnol?language=objc) we should be checking the keys and not the values. 

>The keys for the `serviceCurrentRadioAccessTechnology` dictionary are `NSString` objects, each of which represents a service. The entry associated with a key is nil if the service is not registered on any network.
>
>Although the actual value of a key isn’t important, you can also use it to get the carrier information associated with the service. To do so, pass the key to the `serviceSubscriberCellularProviders` dictionary.

File: GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m [hunk 2]
```diff
@@ -75,7 +75,9 @@ GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() {
   if (@available(iOS 12, *)) {
     NSDictionary<NSString *, NSString *> *networkCurrentRadioAccessTechnologyDict =
         networkInfo.serviceCurrentRadioAccessTechnology;
-    if (networkCurrentRadioAccessTechnologyDict) {
+    if (networkCurrentRadioAccessTechnologyDict.count) {
+      // In iOS 12, multiple radio technologies can be captured. We prefer not particular radio
+      // tech to another, so we'll just return the first value in the dictionary.
       networkCurrentRadioAccessTechnology = networkCurrentRadioAccessTechnologyDict.allValues[0];
```
Comment (mikehaney24) [inline GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m]: This needs to be allKeys, not allValues

Comment (doudounan) [reply GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m]: The keys can be associated with other apis, like serviceSubscriberCellularProviders dictionary, but for the networkCurrentRadioAccessTechnology string we want is stored in the values.

File: GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m (hunk 1 above)
Comment (doudounan) [reply GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m]: Thank you, Morgan. I have changed the if condition to check dictionary.count. And for the networkCurrentRadioAccessTechnology string we want is stored in the values.

**Pull request merged**